src/machine/machine_rp2040_pwm.go

```diff
@@ -256,21 +248,31 @@ func (pwm *pwmGroup) setDivMode(mode uint32) {
 // setPeriod sets the pwm peripheral period (frequency). Calculates DIV_INT and sets it from following equation:
 //  cycles = (TOP+1) * (CSRPHCorrect + 1) * (DIV_INT + DIV_FRAC/16)
 // where cycles is amount of clock cycles per PWM period.
-func (pwm *pwmGroup) setPeriod(period uint64) {
-	targetPeriod := uint32(period)
-	periodPerCycle := getPeriod()
+func (pwm *pwmGroup) setPeriod(period uint32) {
+	periodPerCycle := cpuPeriod()
 	top := pwm.getWrap()
 	phc := pwm.getPhaseCorrect()
-	_, frac := pwm.getClockDiv()
+	// We first modify TOP if desired period is outside divider threshold bounds.
+	limitPeriod := (1 + phc) * periodPerCycle * (1 + top)
+	switch {
```

Comment:
New logic to include bug mentioned in Slack by Fatih. Now modifies Top() value if period is unreachable by modifying just the clock div. Will test tomorrow.

Reply:
Fatih Beşer has confirmed these changes solved the bug!